src/test/java/org/group23/ControllerStructureTest.java

```diff
@@ -123,4 +123,18 @@ public void surveyCreated() throws Exception {
                 .andExpect(MockMvcResultMatchers.view().name("surveyCreated"))
                 .andExpect(MockMvcResultMatchers.model().attributeExists("survey"));
     }
+
+    @Test
+    @DirtiesContext
+    @WithMockUser(username = "user1")
```

Comment:
Could you add a similar test in the security test file testing that an error is returned when we call this endpoint with a different user? The other test methods there should provide good examples of how to check for the error.